src/merge_iterator.rs

```diff
@@ -135,11 +141,22 @@ impl<T1: KeyValueIterator, T2: KeyValueIterator> KeyValueIterator for TwoMergeIt
             if peeked_kv.key != current_kv.key {
                 break;
             }
-            if peeked_kv.seq > current_kv.seq {
+            if peeked_kv.seq > current_kv.seq
```

Comment:
I'm a little curious about `if peeked_kv.seq > current_kv.seq`. I know this code was here before, but it seems odd to me that we're filtering on seq here. The `peek_inner` code appears to return whichever key is lexicographically lower. Couldn't it be the case that it returns KVs in non-seq-ascending order? I'm thinking of something like this:

```
it1=[(k=c, seq=2) => v1, (k=d, seq=5)=> v2]
it2=[(k=a, seq=4) => v1, (k=e, seq=12)=> v2]
```

If I created a TwoMergeIterator, wouldn't peek_inner return a first, then c? And thus, wouldn't c get filtered incorrectly?

Reply:
> If I created a TwoMergeIterator, wouldn't peek_inner return a first, then c? And thus, wouldn't c get filtered incorrectly?

i believe this will not happen because this `if peeked_kv.seq > current_kv.seq` check only happens on the same key. the loop will break as soon as the next key is different with the current one. 

let me append a test case on this case

Reply:
ahh, I missed the `break` call directly above this 😆  Gotcha, then yes, the seq will always apply for the same key and this check makes sense to me. I'll leave this open until you add the test, just for safe keeping.